### ISSUE TITLE
Style/SafeNavigation autocorrection can change behavior with !~ pattern matching

### ISSUE THREAD

**Issue by bgentry** [bbatsov/rubocop#5348]
I had a code block like this:

```ruby
if obj && obj.name !~ /FOO/
  # do something
end
```

If the object is non-nil and its name doesn't match `/FOO/`, then I want to run the code inside the block.

## Expected behavior

Rubocop should probably not change this code, and certainly shouldn't cause the line's functional behavior to change.

## Actual behavior

Rubocop autocorrects under the `Style/SafeNavigation` rule to this:

```ruby
if obj&.name !~ /FOO/
  # do something
end
```

This is functionally different than the original line. That's because `nil !~ /FOO/` actually returns true.

## Steps to reproduce the problem

Run `rubocop -a` on a file containing the original code sample with `Style/SafeNavigation` enabled (as it is by default).

## RuboCop version

```
➜ rubocop -v
0.52.1 (using Parser 2.4.0.2, running on ruby 2.5.0 x86_64-darwin16)
```

## Workaround

If I place parentheses around the second section:

```ruby
if obj && (obj.name !~ /FOO/)
  # do something
end
```

Rubocop does not mess with that line.

**Comment by bbatsov** [comment]
Yeah, it seems we made an ops precedence mistake. @rrosenblum Can you take a look at this?

**Comment by rrosenblum** [comment]
This is related to several other issues that are open for `Style/SafeNavigation` and `Lint/SafeNavigationChain`. The problem is that the cops don't look to see if there is a method chain that makes it ineligible or that will need to be modified during the auto-correction to keep it safe. I'll start working on a fix for this.
